templates/pwm.asg.element.json

```diff
@@ -0,0 +1,884 @@
+                                            "\n"
+                                        ]
+                                    ]
+                                },
+                                "group": "root",
+                                "mode": "000755",
+                                "owner": "root"
+                            }
+                        }
+                    },
+                    "e-finalize": {
+                        "commands": {
+                            "10-signal-success": {
+                                "command": {
+                                    "Fn::Join": [
+                                        "",
+                                        [
+                                            "cfn-signal -e 0 ",
+                                            "   --stack ",
+                                            {
+                                                "Ref": "AWS::StackName"
+                                            },
+                                            "   --resource PWMAutoScalingGroup",
+                                            "   --region ",
+                                            {
+                                                "Ref": "AWS::Region"
+                                            },
+                                            ";sleep 15;reboot now\n"
+                                        ]
+                                    ]
+                                },
+                                "ignoreErrors": "false"
+                            }
+                        }
+                    }
+                }
+            },
+            "Properties": {
+                "BlockDeviceMappings": [
+                    {
+                        "DeviceName": "/dev/sda1",
+                        "Ebs": {
+                            "VolumeSize": 20,
+                            "VolumeType": "gp2",
+                            "DeleteOnTermination": true
+                        }
+                    }
+                ],
+                "IamInstanceProfile": {
+                    "Ref": "PWMInstanceProfile"
+                },
+                "ImageId": {
+                    "Ref": "AmiId"
+                },
+                "InstanceType": {
+                    "Ref": "InstanceType"
+                },
+                "KeyName": {
+                    "Ref": "KeyPairName"
+                },
+                "SecurityGroups": [
+                    {
+                        "Ref": "SecurityGroupIdPWMInstance"
+                    }
+                ],
+                "UserData": {
+                    "Fn::Base64": {
+                        "Fn::Join": [
+                            "",
+                            [
+                                "#!/bin/bash -xe\n\n",
+                                "exec > >(tee /var/log/user-data.log|logger -t user-data -s 2>/dev/console) 2>&1\n",
```

Comment:
We've seen this `exec` stuff cause some pretty strong weirdness before. cloud-init already writes the output to a log file, and we should be capturing that output in cloudwatch-logs. this approach may be OBE at this point